### DISCUSSION TITLE
What does the new draft policy mean for rust-bitcoin?

### DISCUSSION THREAD

**Discussion by sanket1729** [rust-bitcoin/rust-bitcoin#1778] [General]
I am not a lawyer, but does this mean we need to rename all our crates?

form: https://docs.google.com/forms/d/e/1FAIpQLSdaM4pdWFsLJ8GHIUFIhepuq0lfTg_b0mJ-hvwPdHa4UTRaAg/viewform

and associated "policy": https://docs.google.com/document/d/1ErZlwz9bbSI43dNo-rgQdkovm2h5ycuW220mWSOAuok/edit#

I don't think we should put too much time into this because its still a draft, but I hope it does not get to that. FWIW, I submitted my feedback.

**Comment by TheBlueMatt** [comment]
Given the outcry of hate I'm gonna bet it won't go through, but, yes, I believe it would mean that.

**Comment by apoelstra** [comment]
Lol. Yeah, I don't see this going through, but I'm sure it wall cause even more irreparable damage to the ecosystem regardless.

**Comment by tcharding** [comment]
WTF! this `rust-bitcoin` crate is not officially endorsed by the Rust language team - wish you guys told me that before I started working on it, I suppose your `rust-bitcoin` organisation is a phony too, we should change the name to `bitcoin-rs` - oh wait that is taken, what are the odds (read: sarcasm).

**Reply by Harshil-Jani** [reply]
How about crab-bitcoin XD https://crablang.org/
This sounds so weird anyway :)

**Comment by apoelstra** [comment]
So, FWIW our **crate names** actually don't have `rust-` prefixes in them, just the repo names. So this may not be as disruptive as we expect.

Though this whole thing is still stupid enough to make me suspect deliberate sabotage of the open-source Rust ecosystem.

**Comment by Kixunil** [comment]
Heh, wanted to open this as well, didn't have much time and forgot. Yeah, I've seen some people complaining so there's a good chance we don't have to rename anything. Also we have domain name too. I wonder if just grabbing the names before the new policy was created is enough to void their claims.

**Comment by junderw** [comment]
No use discussing until a policy is passed.

It's not like the moment the policy passes we'll all magically be sent to jail.

They will have to pay lawyers to write cease and desist letters to all the crates with rust in it first...

I personally think that this should only be discussed once they have made it policy and there are reports of Rust crates receiving c&d letters.

**Reply by apoelstra** [reply]
Yes, agreed. And also yes -- *assuming* this is even passed, which I highly doubt, they will send us a C&D before any other action. In that case we'll post the C&D on our website, since I'd consider this a mark of pride for the project, and we'll also have to change the project's name, which is a bit annoying but not much. Note that none of our crates actually use `rust`, only our repo name and our domain name, both of which are easy to swap out, especially in this pre-1.0 era when we don't have many users we don't know.

But realistically it looks like the most likely outcome is that the policy will be modified to no longer be so restrictive toward Rust projects, and for our purposes it'll be a total nothingburger.

**Reply by Kixunil** [reply]
Was still a good thing to spread awareness about the form (which is now closed).

Anyway I'm sooo tempted to add "Supported by The Rust Foundation... LOL, no, why would you think that?!" or a similar sarcastic note on the website... :laughing:

**Reply by apoelstra** [reply]
I think it would be funny to clarify that *they're* not affililiated with *us*, since we're a software project and they're random busybodies.

But probably this sort of snark (or sarcasm) would not be productive :)

**Reply by dpc** [reply]
I heard that `rust-bitcoin` was started in https://en.wikipedia.org/wiki/Rust,_Baden-W%C3%BCrttemberg , wasn't it?
